backend/game/models/game.py

```diff
@@ -0,0 +1,44 @@
+from django.db import models
+from game.models import Player
+
+
+class GameManager(models.Manager):
+    """Custom manager for Game, overwrites create method"""
+    def create(self, creator, name, password):
+        """Create a new game, and a player admin associated with creator
+
+        Args:
+            creator: users.User, User which creates the new game
+            name: string, game name (optional, set to "creator's game" set to None)
+            password: string, game password (no password if None)
+        """
+        # Set a default name if not provided
+        if name is None:
+            name = str(creator.username) + "'s game"
+
+        new_game = super().create(name=name, password=password)
+
+        # creator becomes a player and admin of new game
+        Player.objects.create(game=new_game, user=creator, is_admin=True)
+        return new_game
+
+    def with_user(self, user_id):
+        """Returns all games with a player controlled by user having id=user_id"""
+        ids_with_user = []
+        for game in self.all():
+            # if game has a player controlled by user
+            if game.players.filter(user__id=user_id):
+                ids_with_user.append(game.id)
+        return self.filter(id__in=ids_with_user)
```

Comment:
Jolie fonction ! Je pense qu'il vaut mieux utiliser une liste par compréhension plutôt qu'une boucle for :
`ids_with_user = [player.game.id for player in Player.objects.filter(user_id=user_id)]`
`return self.filter(id__in=ids_with_user)`

Reply:
Pour ce genre de fonction relativement isolée et un peu compliquée je pense que ça vaut le coup d'écrire un test unitaire